src/hip_memory.cpp

```diff
@@ -2137,18 +2137,35 @@ hipError_t hipIpcOpenMemHandle(void** devPtr, hipIpcMemHandle_t handle, unsigned
         // Get the current device agent.
         hc::accelerator acc;
         hsa_agent_t* agent = static_cast<hsa_agent_t*>(acc.get_hsa_agent());
-        if (!agent) return hipErrorInvalidResourceHandle;
-
-        ihipIpcMemHandle_t* iHandle = (ihipIpcMemHandle_t*)&handle;
-        // Attach ipc memory
-        auto ctx = ihipGetTlsDefaultCtx();
-        {
-            LockedAccessor_CtxCrit_t crit(ctx->criticalData());
-            // the peerCnt always stores self so make sure the trace actually
-            hsa_status_t hsa_status = hsa_amd_ipc_memory_attach(
-                (hsa_amd_ipc_memory_t*)&(iHandle->ipc_handle), iHandle->psize, crit->peerCnt(),
-                crit->peerAgents(), devPtr);
-            if (hsa_status != HSA_STATUS_SUCCESS) hipStatus = hipErrorMapBufferObjectFailed;
+        if (!agent) {
+            hipStatus = hipErrorInvalidResourceHandle;
+        } else {
+            ihipIpcMemHandle_t* iHandle = (ihipIpcMemHandle_t*)&handle;
+            // Attach ipc memory
+            auto ctx = ihipGetTlsDefaultCtx();
+            {
```

Comment:
Nit: Why is this block {..} needed when there is nothing after it?

Reply:
This is to understand scope of LockedAccessor_CtxCrit_t.